File: Python/1137.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n < 2:
            return n
        
        first = 0
        second = 1
        third = 1
        for i in range(n - 2):
            tmp = first
            first = second
            second = third
            third = tmp + first + second
        
        return third

File: C++/1137.cpp
class Solution {
public:
    int tribonacci(int n) {
        int T[38] = {};
        T[1] = T[2] = 1;
        for (int i = 3; i <= n; ++i) {
            T[i] = T[i - 1] + T[i - 2] + T[i - 3];
        }
        return T[n];
    }
};
